wb/main/huggingface_api/huggingface_api.py

```diff
@@ -89,29 +92,69 @@ def json(self):
         }
 
 
+supports_classification = {
+    model_type for model_type, tasks in FeaturesManager._SUPPORTED_MODEL_TYPE.items()
+    if 'sequence-classification' in tasks
+}
+
 contains_decoder = {
-        model_type for model_type, tasks in FeaturesManager._SUPPORTED_MODEL_TYPE.items()
-        if any("with-past" in task for task in tasks)
-    }
+    model_type for model_type, tasks in FeaturesManager._SUPPORTED_MODEL_TYPE.items()
+    if any('with-past' in task for task in tasks)
+}
+
+
+def get_vocab_filenames(model_type: str) -> List[Set[str]]:
+    files = []
+    for tokenizer_type in ('', '_fast'):
+        try:
+            module = importlib.import_module(
+                f'.tokenization_{model_type}{tokenizer_type}', f'transformers.models.{model_type}'
+            )
+        except ModuleNotFoundError:
+            continue
+        vocab_dict = getattr(module, 'VOCAB_FILES_NAMES')
+        if vocab_dict:
+            files.append(set(vocab_dict.values()))
+    return files
+
+
+supported_models = supports_classification.difference(contains_decoder)
+
+tokenizer_vocab_files = {
+    model_type: get_vocab_filenames(model_type) for model_type in supported_models
+}
+
+
+def has_missing_tokenizer_files(model: ModelInfo) -> bool:
+    repo_files = {file.rfilename for file in model.siblings}
+    tokenizers_files = tokenizer_vocab_files[model.config['model_type']]
+    return all(tokenizer_files - repo_files for tokenizer_files in tokenizers_files)
 
 
 def _validate_hf_model(model: ModelInfo) -> ValidationResult:
```

Comment:
Let's move all validation logic in a separate file under `huggingface_api` module

Reply:
Agree. Moved validation to separate files.